src/tests/unit/test_rich_rule.py

```diff
@@ -24,7 +24,7 @@ def _exp_rule_check_2(exp_rule, rule):
         == 'rule family="ipv6" source address="1:2:3:4:6::" icmp-block name="redirect" log prefix="redirected: " level="info" limit value="4/m"'
     )
     s = rule.element
-    assert type(s) == firewall.core.rich.Rich_IcmpBlock
+    assert type(s) is firewall.core.rich.Rich_IcmpBlock
```

Comment:
This warning happens because latest `pip install flake8 -U` gives you 7.0.0, while the github action seems to get something older. That's somewhat surprising, because we don't set a fixed version in pip (which we should).

This is of course a recurring topic. I also run Fedora 39's `black==23.7.0`, which formats some code differently than the blessed `black==22.12.0`. But luckily, I can find a format that works with both versions.

May point is, very soon we shouldupdate the github action to use newer `black`/`flake8`... I will open an PR for that.

Reply:
hm... while it makes sense to tie `black==22.12.0` (as it affects the chosen coding style), it makes a bit less sense to tie `flake8==7.0.0`. Nothing to do here.

I am only surprised that `pip install flake8` on github doesn't give 7.0.0. Dunno...